src/lib/apiTypesV2/Subscription.h

```diff
@@ -0,0 +1,78 @@
+#ifndef SRC_LIB_APITYPESV2_SUBSCRIPTION_H
+#define SRC_LIB_APITYPESV2_SUBSCRIPTION_H
+
+/*
+*
+* Copyright 2015 Telefonica Investigacion y Desarrollo, S.A.U
+*
+* This file is part of Orion Context Broker.
+*
+* Orion Context Broker is free software: you can redistribute it and/or
+* modify it under the terms of the GNU Affero General Public License as
+* published by the Free Software Foundation, either version 3 of the
+* License, or (at your option) any later version.
+*
+* Orion Context Broker is distributed in the hope that it will be useful,
+* but WITHOUT ANY WARRANTY; without even the implied warranty of
+* MERCHANTABILITY or FITNESS FOR A PARTICULAR PURPOSE. See the GNU Affero
+* General Public License for more details.
+*
+* You should have received a copy of the GNU Affero General Public License
+* along with Orion Context Broker. If not, see http://www.gnu.org/licenses/.
+*
+* For those usages not covered by this license please contact with
+* iot_support at tid dot es
+*
+* Author: Orion dev team
+*/
+
+#include <string>
+#include <vector>
+
+#include "../ngsi/Duration.h"
+#include "../ngsi/Throttling.h"
+
+struct EntID
+{
+  std::string id;
+  std::string idPattern;
+  std::string type;
+  std::string toJson();
+};
```

Comment:
Sure, it may make sense to have a different class for different uses of Entity, and all other concepts/classes also.
It adds to the number of lines of the project, but it makes it faster and probably easier to read too.
Some parts of the code will be duplicated but that's not really a problem.

NTC